base/reflection.jl

```diff
@@ -1053,6 +1054,8 @@ methods(@nospecialize(f), @nospecialize(t), mod::Module) = methods(f, t, (mod,))
 function methods_including_ambiguous(@nospecialize(f), @nospecialize(t))
     tt = signature_type(f, t)
     world = get_world_counter()
+    (ccall(:jl_is_in_pure_context, Bool, ()) || world == typemax(UInt)) &&
```

Comment:
Why the `ccall(:jl_is_in_pure_context, Bool, ())` here? Shouldn't the check for `world=typemax` be sufficient? Also, when using GeneratedFunctionStub + world argument, if we'd add a world argument to `methods_including_ambiguous` it should be fine to call that function, right?

Reply:
There are no edges computed in this function, so it is illegal to use from pure contexts

The typemax check is to throw a friendly error rather than letting it get to the backend and expect the user to know what to do

Reply:
When do we have `get_world_counter() == typemax(UInt)` without `jl_is_in_pure_context`, or vice versa?

Reply:
They are fundamentally different things, passed from different places

Reply:
Could you elaborate for my understanding then? I only ever saw `jl_is_in_pure_context == true` in generated functions, which  now have `get_world_counter() == typemax(UInt)`.